### ISSUE TITLE
Binary focal loss is numerically unstable

### ISSUE THREAD

**Issue by zimka** [kornia/kornia#2115]
### Describe the bug

Binary Focal Loss (kornia.losses.focal.binary_focal_loss_with_logits) is numerically unstable.
It looks like a regression bug - issue reproduces with kornia==0.6.9 but does not with kornia==0.6.4.

It probably happens because of `y = (1 - target)* log(sigmoid(-logit))` that happens for prob neg [here](https://github.com/kornia/kornia/blob/v0.6.9/kornia/losses/focal.py#L217)

When `logit->inf`:
* sigmoid(-logit) -> 0
* log(sigmoid(-logit)) -> log(0) -> (-inf)
* (1-target) == 0
* y = 0 * (-inf) = nan

### Reproduction steps

```bash
1. `import torch; from kornia.losses.focal import binary_focal_loss_with_logits`
2. `binary_focal_loss_with_logits(torch.tensor(100.).view(1, 1, 1).float(), torch.tensor(1.).view(1, 1, 1).float()).item()`

Returns nan for kornia==0.6.9, though it should be 0 (positive class=1 with huge positive logit=100)
```

### Expected behavior

Expect to get 0 value in the example instead of nan.

### Environment

```shell
wget https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py
# For security purposes, please check the contents of collect_env.py before running it.
python collect_env.py
```
- PyTorch Version (e.g., 1.0): 1.12.1
- OS (e.g., Linux): Linux
- How you installed PyTorch (`conda`, `pip`, source): conda
- Build command you used (if compiling from source):
- Python version: 3.9.0
- CUDA/cuDNN version:
- GPU models and configuration: CPU
- Any other relevant information:
```


### Additional context

_ I have checked that both on Windows and Linux for kornia==0.6.4 issue does not happen, which probably means that it depends only on kornia but not on third-party libs or env_

**Comment by shijianjian** [comment]
Probably we need to limit the lower bound for:

https://github.com/kornia/kornia/blob/d6882a24235839ebd5b82bf0993115be5371c990/kornia/losses/focal.py#L212-L213

@zimka Can you try fix it then throw a PR?

**Comment by zimka** [comment]
PR that fixes the issue https://github.com/kornia/kornia/pull/2125
@shijianjian FYI
